pkg/inventory/inventory-client.go

```diff
@@ -421,3 +431,40 @@ func (cic *ClusterInventoryClient) ApplyInventoryNamespace(obj *unstructured.Uns
 func (cic *ClusterInventoryClient) getMapping(obj *unstructured.Unstructured) (*meta.RESTMapping, error) {
 	return cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind(), obj.GroupVersionKind().Version)
 }
+
+func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured, dryRun common.DryRunStrategy) error {
+	if dryRun.ClientOrServerDryRun() {
+		klog.V(4).Infof("dry-run update inventory status: not updated")
+		return nil
+	}
+	status, found, _ := unstructured.NestedMap(obj.UnstructuredContent(), "status")
+	if !found {
+		return nil
+	}
+
+	klog.V(4).Infof("update inventory status")
+	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return err
+	}
+	resource := cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace())
+	meta := metav1.TypeMeta{
+		Kind:       obj.GetKind(),
+		APIVersion: obj.GetAPIVersion(),
+	}
+	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta}, "status")
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof("skip updating inventory subresource status for %v: %v", meta, err)
+			return nil
+		}
+		return err
+	}
+
+	err = unstructured.SetNestedMap(liveObj.UnstructuredContent(), status, "status")
+	if err != nil {
+		return err
+	}
+	_, err = resource.UpdateStatus(context.TODO(), liveObj, metav1.UpdateOptions{TypeMeta: meta})
+	return err
```

Comment:
fmt.Errorf("failed to write updated inventory status to cluster: %w", err)

Reply:
done